src/share/cxx/Elements.cpp

```diff
@@ -125,6 +142,35 @@ void Elements::init_2d(CF90Ptr &D, CF90Ptr &Dinv, CF90Ptr &fcor,
       }
     }
   }
+  
+  if(!consthv){
```

Comment:
Would be good to make these next two blocks of loops functions in SyncUtils.hpp, if suitable ones don't already exist.

Reply:
ok, will do.